python/ray/tune/tests/test_trial.py

```diff
@@ -2,14 +2,61 @@
 import sys
 import pytest
 
+from ray.exceptions import RayActorError, RayTaskError
 from ray.train import Checkpoint
+from ray.train.constants import RAY_TRAIN_COUNT_PREEMPTION_ERRORS
 from ray.train._internal.session import _TrainingResult
 from ray.train._internal.storage import StorageContext
 from ray.tune.experiment import Trial
 
 from ray.train.tests.util import mock_storage_context
 
 
+@pytest.fixture
+def trial(tmp_path):
+    yield Trial(
+        "mock",
+        stub=True,
+        storage=mock_storage_context(storage_path=str(tmp_path)),
+    )
+
+
+@pytest.mark.parametrize("count_preemption_errors", [False, True])
+def test_handle_preemption_error(
+    trial: Trial, count_preemption_errors: bool, monkeypatch
+):
+    """Check that the Trial counts preemption errors correctly."""
+    if count_preemption_errors:
+        monkeypatch.setenv(RAY_TRAIN_COUNT_PREEMPTION_ERRORS, "1")
+
+    num_errors_before = trial.num_failures
```

Comment:
Any reason this isn't just 0?

Reply:
Ah I can change it. I was worried about someone changing this trial fixture and impacting this test accidentally. but i think it's just more confusing.